test/test_utils.jl

```diff
@@ -105,6 +107,21 @@ function test_FiniteDiff(kernelfunction, args = nothing, dims = [3, 3])
                     testfunction(kernelfunction(p), A, B, dim)
                 end
             end
+
+            @test_nowarn gradient(:FiniteDiff, A) do a
+                testdiagfunction(k, a, dim)
+            end
+            @test_nowarn gradient(:FiniteDiff , A) do a
+                testdiagfunction(k, a, B, dim)
+            end
+            @test_nowarn gradient(:FiniteDiff, B) do b
+                testdiagfunction(k, A, b, dim)
+            end
+            if !(args === nothing)
```

Comment:
It doesn't really matter but it feels a bit more natural to write:
```suggestion
            if args !== nothing
```